argo_workflow_tools/dsl/workflow.py

```diff
@@ -17,7 +17,7 @@ def __init__(
         self,
         name: str,
         entrypoint: Callable,
-        arguments: Dict[str, str] = None,
```

Comment:
it should be a Union[Dict[str, str],List[Union[argo.Arguments, argo.Parameter]]].
or be shortened with type alias

Reply:
Union[Dict[str, Any], List[Union[argo.Artifact, argo.Parameter]]]

Reply:
I've fixed it now @thegreymatter  :)